articles/go-sdk.html

```diff
@@ -83,7 +81,8 @@ <h3>The Simplest Go Program</h3>
 
 <p>
 The words <code>package</code> and <code>func</code> are two keywords.
-The two <code>main</code> words are two identifiers.
+The two <code>main</code> words are two identifiers,
+and <b>main</b> is a special predefined identifier.

```

Comment:
In fact, "main" is not predefined.

Reply:
Hm - isn't this just another way to say same?

A package name spelled 'm' 'a' 'i' 'n' is recocgnised (as being an identifier previously defined by and thus known to Go), the language gives it a special meaning and the toolchain gives it special treatment accordingly.

A (niladic) package-level function spelled 'm' 'a' 'i' 'n'? Some story.

`init()` is another example.

`vendor` and `internal` are predefined identifieres in the context of nodes of import paths ...

btw: I've never seen a comprehensive list of these 'magic words' anywhere - and such might make a great addition to Your work.

Reply:
I think, a "predefined" identifier means the identifier is defined somewhere already.
For example, `int`, `bool`, `true` and `iota` etc are defined in the `builtin` package.
But we can't the places where `main` is defined.
so it is a rule instead of a predefined thing.

Reply:
And we can use the `true` identifier directly, but we can't call `main()` directly without declaring it by ourselves.

Reply:
"defined somewhere already" - yes!

Just: (some)where? ???
The parser/lexer 'understands' it?
=> it's called a `Keyword` (as far as I know)
else: it's (someones) literal elsewhere/later (in the tool chain).

step aside: the `builtin` package is nothing but a VERY clever way do keep `godoc` happy.

As You praise in Your intro, Go is intentionally simple!
As simple as possible! And no simpler than that
(Albert Einstein coined the phrase ..)

And their team is rightfully happy & proud to have **only** following 5*5 keywords in their `doc/go_spec.html`:

break        default      func         interface    select
case         defer        go           map          struct
chan         else         goto         package      switch
const        fallthrough  if           range        type
continue     for          import       return       var

Wonders of wonders: none of the words You mentioned in Your samples is among them.

But later down the chain, someone looks at some AST-node (and it's type) and says:
Oh - a type! named 'int' :-) Beautiful, I know how to make him happy! :-)
Oh - a package! named 'main' :-) Beautiful, I know how to handle this :-)
    ( and I'll make sure to find a function of same name which I'll be more than happy and honored to give the unique privilege of becoming the programs entry point.)
\<aside>Well - it's in progress - there is an issue about details in a recent GoWeekly \</aside>

LT;DR iI t's not a KeyWord - it's been given a special meaning (and You cannot even overrule this!) - well - then it's called -as I said.

That's what I know.

Reply:
Having said all this - never ever did I find any (nearly) complete documentation of such special literals.

This knowledge is spread all over the place, and cries for consolidation!

Your book might be a great place to provide such! :-)

I'd be in ...

Reply:
Ah, I'm still not persuaded.
I still think predefined identifiers can be used without declaring them by ourselves. 
We can't do the following
```golang
package pkg

import "main" // cannot find package "main" 

func f() {
     main() // undefined: main
}
```